### PR TITLE
Implemented Mac Address type #72

### PR DESCRIPTION
**Pull request by emir-sehovic1**

### CODE REVIEW

File: tests/Nox.Types.Tests/Types/MacAddress/MacAddressTests.cs [hunk 1]
```diff
@@ -3,9 +3,117 @@ namespace Nox.Types.Tests.Types;
 
 public class MacAddressTests
```
Comment (dejobratic) [inline tests/Nox.Types.Tests/Types/MacAddress/MacAddressTests.cs:4]: According to the latest info, test should be written using FluentAssertions. Can you please add the required changes here? 

https://regusit.atlassian.net/wiki/spaces/NOX/pages/3153264693/Test+Projects

You can find sample implementation here: https://github.com/NoxOrg/Nox.Generator/blob/main/tests/Nox.Types.Tests/Types/Volume/VolumeTests.cs

File: src/Nox.Types/Types/MacAddress/MacAddress.cs [hunk 2]
```diff
@@ -1,9 +1,116 @@
...
+
 namespace Nox.Types;
 
+/// <summary>
+/// Represents a Nox <see cref="MacAddress"/> type and value object.
+/// </summary>
+/// <remarks>Placeholder, needs to be implemented</remarks>
+public sealed class MacAddress : ValueObject<string, MacAddress>
+{
+    private static readonly int MacAddressLengthInBytes = 6;
```
Comment (dejobratic) [inline src/Nox.Types/Types/MacAddress/MacAddress.cs:15]: Can we change this to private const int?

File: src/Nox.Types/Types/MacAddress/MacAddress.cs [hunk 3]
```diff
@@ -1,9 +1,116 @@
...
+
+    public new static MacAddress From(string value)
```
Comment (dejobratic) [inline src/Nox.Types/Types/MacAddress/MacAddress.cs:17]: Can you please add comments on public methods?

**Review by dejobratic — CHANGES_REQUESTED** [review]
Looks good to me. Added some minor comments.

**Review by dejobratic — DISMISSED** [review]
Looks good. Thanks.

File: src/Nox.Types/Types/MacAddress/MacAddress.cs [hunk 4]
```diff
@@ -1,9 +1,152 @@
...
+using System.Net.NetworkInformation;
+using System;
+using System.Globalization;
+
 namespace Nox.Types;
 
+/// <summary>
+/// Represents a Nox <see cref="MacAddress"/> type and value object.
+/// </summary>
+/// <remarks>Placeholder, needs to be implemented</remarks>
```
Comment (rochar) [inline src/Nox.Types/Types/MacAddress/MacAddress.cs:12]: you can remove the remark

**Review by rochar — APPROVED** [review]
looks good, just a small typo

File: src/Nox.Types/Types/MacAddress/MacAddress.cs [hunk 5]
```diff
@@ -1,9 +1,152 @@
...
+{
+    private const int MacAddressLengthInBytes = 6;
+
+    /// <summary>
+    /// Creates a new instance of <see cref="MacAddress"/> object.
+    /// </summary>
+    /// <param name="value">The string value to create the <see cref="MacAddress"/> with</param>
+    /// <returns></returns>
+    /// <exception cref="TypeValidationException"></exception>
+    public new static MacAddress From(string value)
```
Comment (rochar) [inline src/Nox.Types/Types/MacAddress/MacAddress.cs:23]: can you rename the property value to something meaningful like macAddressText

File: src/Nox.Types/Types/MacAddress/MacAddress.cs [hunk 6]
```diff
@@ -1,9 +1,152 @@
...
+        return newObject;
+    }
+
+    /// <summary>
+    /// Creates a new instance of <see cref="MacAddress"/> object.
+    /// </summary>
+    /// <param name="value">The ulong value to create the <see cref="MacAddress"/> with</param>
+    /// <returns></returns>
+    /// <exception cref="TypeValidationException"></exception>
+    public static MacAddress From(ulong value)
```
Comment (rochar) [inline src/Nox.Types/Types/MacAddress/MacAddress.cs:48]: same here, value has no good meaning

**Review by rochar — CHANGES_REQUESTED** [review]

**Review by rochar — APPROVED** [review]

**Pull request merged**